src/pathfinder/yapf/nodelist.hpp

```diff
@@ -15,38 +15,28 @@
 
 /**
  * Hash table based node list multi-container class.
- *  Implements open list, closed list and priority queue for A-star
- *  path finder.
+ *  Implements open list, closed list and priority queue for A-star pathfinder.
  */
-template <class Titem_, int Thash_bits_open_, int Thash_bits_closed_>
-class CNodeList_HashTableT {
+template <class Titem, int Thash_bits_open, int Thash_bits_closed>
+class NodeList {
 public:
-	typedef Titem_ Titem;                                        ///< Make #Titem_ visible from outside of class.
-	typedef typename Titem_::Key Key;                            ///< Make Titem_::Key a property of this class.
-	using CItemArray = std::deque<Titem_>;                       ///< Type that we will use as item container.
-	typedef HashTable<Titem_, Thash_bits_open_  > COpenList;     ///< How pointers to open nodes will be stored.
-	typedef HashTable<Titem_, Thash_bits_closed_> CClosedList;   ///< How pointers to closed nodes will be stored.
-	typedef CBinaryHeapT<Titem_> CPriorityQueue;                 ///< How the priority queue will be managed.
+	using Item = Titem;
```

Comment:
Is that a good idea? the 'T' prefix is there for a reason...

(ok, it's not there for the Key typedef, but I guess that's not actually a template param)

Reply:
It's not really a template parameter anymore if you use a typedef / alias, and our style guide doesn't specifically describe what to do in this case. IMO the non-prefixed version makes more sense, but this is a bit of a grey area.